flexmeasures/ui/crud/assets.py

```diff
@@ -278,11 +278,22 @@ def post(self, id: str):
         else:
             asset_form = with_options(AssetForm())
             if not asset_form.validate_on_submit():
+                asset = Asset.query.get(id)
+                latest_measurement_time_str, asset_plot_html = get_latest_power_as_plot(
+                    asset
+                )
+                # Display the form data, but set some extra data which the page wants to show.
+                asset_info = asset_form.data.copy()
+                asset_info["id"] = id
+                asset_info["owner_id"] = asset.owner_id
+                asset_info["entity_address"] = asset.entity_address
                 return render_flexmeasures_template(
-                    "crud/asset_new.html",
+                    "crud/asset.html",
                     asset_form=asset_form,
+                    asset=asset_info,
```

Comment:
Why not hand over the `asset` object itself to Jinja? Then it would be easier to adjust asset.html (e.g. show a new asset property) without the need to also explicitly add it to the `asset_info` dict.

Reply:
The task at hand here is to give the form data to Jinja, so that invalid data can be properly displayed (for the user to be able to adjust it). The asset object is fresh from the database, not what the user just entered.